### PR TITLE
Refactor/card metadata

### PR DESCRIPTION
**Pull request by thorrester**
### Short Summary
Reduces clutter in `ModelCard` and `DataCard` attributes

### Context
`ModelCard` and `DataCard` have too many attributes that can more broadly be defined as `metadata`. This PR introduces a `ModelCardMetadata` and `DataCardMetadata` class that are used in the new attr `metadata`.  This will allow both model and data cards to have a simpler public interface

### Is this a Breaking Change?
Could be. The attrs that have been moved to `metadata` are attrs that are not regularly accessed or used by users. However, because the public interface is changing, we will need to announce it when the release is made for v1.5. 2 additional tests are added to ensure backward compat on <V1.5

### CODE REVIEW

File: opsml/model/types.py [hunk 1]
```diff
@@ -129,6 +129,25 @@ class ModelReturn(BaseModel):
     model_config = ConfigDict(frozen=False, protected_namespaces=("protect_",))
```
Comment (thorrester) [inline opsml/model/types.py:131]: Moved some files around

File: opsml/app/routes/data.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ def download_data_profile(
         storage_client = request.app.state.storage_client
         return StreamingResponse(
             storage_client.iterfile(
```
Comment (thorrester) [inline opsml/app/routes/data.py:47]: adding `metadata` touches a lot of files

File: opsml/registry/cards/types.py [hunk 3]
```diff
@@ -100,4 +82,143 @@ class RunCardArgs(str, Enum):
     PIPELINE_UID = "pipelinecard_uid"
 
 
+class Description(BaseModel):
+    summary: Optional[str] = None
+    intended_use: Optional[str] = None
+    sample_code: Optional[str] = None
+    Notes: Optional[str] = None
+
+
```
Comment (thorrester) [inline opsml/registry/cards/types.py:91]: `DataCardUris` and `ModelCardUris` already exist. Moved them around so everything was in one place

File: tests/test_registry/test_backward.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
+from typing import Any, Dict, Optional
```
Comment (damonallison) [inline tests/test_registry/test_backward.py:1]: 👍🏻

**Review by damonallison — APPROVED** [review]

**Pull request merged**